zgw_consumers/admin.py

```diff
@@ -33,5 +34,16 @@ class ListZaaktypenMixin:
 
     def formfield_for_dbfield(self, db_field, request, **kwargs):
         if db_field.name in self.zaaktype_fields:
-            return get_zaaktype_field(db_field, request, **kwargs)
+            try:
+                return get_zaaktype_field(db_field, request, **kwargs)
+            except ClientError:
+                messages.add_message(
+                    request,
+                    messages.ERROR,
+                    _(
+                        "The credentials entered for one or more of the API services are invalid. "
+                        "Please check the client id as well as the secret and try again."
+                    ),
+                )
+                return None
```

Comment:
this is not entirely correct. You could also be getting an HTTP 400 because incorrect querystring parameters are being used, which is different from an HTTP 403.

This hardcoded message completely ignores the information present in the error response itself. You should be extracting that and showing it to the end user, e.g.:

```
Failed to retrieve available zaaktypen (got {http_status} - {error_detail}).
```

These new lines of code are also not tested, see the missing code coverage.